### ISSUE TITLE
Some hotkeys not working 

### ISSUE THREAD

**Issue by caldertron** [sabrogden/Ditto#47]
Hello, 
My co-workers and I use ditto quite often in our daily work and we love and appreciate it so much. A colleague and I both use the same assignment scheme for our hotkeys - Ctrl+F1, Ctrl+F2, on down the line. We both have consistently found that items assigned to CTRL+F7 and CTRL+F8 do not paste the text assigned in Ditto proper, and are looking for troubleshooting suggestions. 

For example, let's say I've assigned 5 lines of lorem ipsum to CTRL+F7, and a subsequent 5 lines to CTRL+F8. Often CTRL+F7 will paste correctly, but CTRL+F8 will paste nothing. Sometimes the opposite will be true (CTRL+F8 will work and CTRL+F7 will not), but the problem is usually always with the same keys and the same chunk of text.

**- If I toggle to the clipboard manager and click on the saved entry, it pastes all lines properly.**
- Sometimes it will only paste one or two lines. 
- There does not seem to be a pattern of when it works and when it does not. 
- There are no conflicting hotkeys that I have been able to identify in Windows or in the browser we often work from.
- If I then try to assign that same lines of text to say CTRL+F9, it will also not paste properly. I've even tried assigning it to different hotkeys that didn't use either CTRL or one of the function keys and have consistently had the same issue. This is most perplexing because again, if I toggle to the clipboard manager and click on the saved entry, it pastes all lines properly.
- If I try to assign new text to a CTRL+F7/F8 hotkey that previously wasn't working, it also will not paste. 
- Sometimes after toggling the hotkey it will appear in the most recently-pasted items, but sometimes it will not

Any suggestions or solutions would be very appreciated!

**Comment by sabrogden** [comment]
Are you assigning global hotkeys to individual clips (clip properties, set hot key, check 'Hotkey available globally), as opposed to global position(index 1, 2, 3...) hotkeys.

Let's start by right clicking on the taskbar icon - Global Hotkeys, does it show that they are all registered.

**Comment by caldertron** [comment]
Yes, we are assigning hotkeys using the clip properties -> set hotkey -> hotkey available globally. When I go to the Global Hotkeys, CTRL+F7 and CTRL+F8 both show as registered hotkeys and the description matches the chosen text.

**Comment by sabrogden** [comment]
Next, we need to see if Ditto is getting notified when you press the hotkey.  We need to enable debug logging to see this, https://github.com/sabrogden/Ditto/wiki/Debug-Logging, press ctrl-d and ctrl-f back to back really fast, this could be easier, sorry.  This will write to the file c:\Users\<user>\AppData\Roaming\Ditto\ditto.log.  When you press a hotkey the line 

`Pasting clip from global shortcut, clipId: ....`

will be logged, do you see this line when pressing ctrl-f7 or ctrl-f8

One thing that confused me when I was testing this was that once you press ctrl-f1 and that pastes a clip you have to release the control key before another shortcut key will work (so you can't hold down the control key and keep pressing f1, f2, f3 to paste), could this be happening for you.

**Comment by caldertron** [comment]
Complication - I'm on an administrated machine and don't have access to everything/things are in weird places. I followed your instructions for debug logging, but I haven't been able to locate a ditto.log file anywhere on my computer. 

We are definitely releasing ctrl between pasting - we have to enter non-pasteable information in between each pastable bit.

**Comment by caldertron** [comment]
Okay, I was able to find the log!! Here's what happens after I press CTRL+F7 (works) and then CTRL+F8 (does not work) --

[2021/1/26 15:40:04.394 - ClipboardViewer.cpp 242] OnClipboardChange - Start
[2021/1/26 15:40:04.418 - ClipboardViewer.cpp 324] INCLUDE app names: *, Active App: notepad.exe
[2021/1/26 15:40:04.425 - ClipboardViewer.cpp 337] Inlclude app names Found Match * - notepad.exe
[2021/1/26 15:40:04.432 - ClipboardViewer.cpp 267] OnDrawClipboard:: *** SetTimer *** -17494578
[2021/1/26 15:40:04.464 - ClipboardViewer.cpp 244] OnClipboardChange - End
[2021/1/26 15:40:04.578 - ClipboardViewer.cpp 396] OnDrawClipboard::OnTimer -17494437
[2021/1/26 15:40:04.580 - CopyThread.cpp 56] OnClipboardChange - Start
[2021/1/26 15:40:04.581 - CopyThread.cpp 109] LoadFromClipboard - Before
[2021/1/26 15:40:04.584 - Clip.cpp 455] Tried to set description from cf_unicode text, Set: 1, Desc: [[2021/1/26 15:38:06.121 - Even]
[2021/1/26 15:40:04.585 - Clip.cpp 487] Begin enumerating over supported types, Count: 6
[2021/1/26 15:40:04.587 - Clip.cpp 501] Begin try and load type CF_TEXT
[2021/1/26 15:40:04.588 - Clip.cpp 558] End of load - type CF_TEXT, Success: 1
[2021/1/26 15:40:04.590 - Clip.cpp 501] Begin try and load type Rich Text Format
[2021/1/26 15:40:04.594 - Clip.cpp 511] End of load - Data is not available for type Rich Text Format
[2021/1/26 15:40:04.595 - Clip.cpp 501] Begin try and load type CF_UNICODETEXT
[2021/1/26 15:40:04.597 - Clip.cpp 558] End of load - type CF_UNICODETEXT, Success: 1
[2021/1/26 15:40:04.600 - Clip.cpp 501] Begin try and load type CF_HDROP
[2021/1/26 15:40:04.602 - Clip.cpp 511] End of load - Data is not available for type CF_HDROP
[2021/1/26 15:40:04.604 - Clip.cpp 501] Begin try and load type CF_DIB
[2021/1/26 15:40:04.606 - Clip.cpp 511] End of load - Data is not available for type CF_DIB
[2021/1/26 15:40:04.608 - Clip.cpp 501] Begin try and load type HTML Format
[2021/1/26 15:40:04.611 - Clip.cpp 511] End of load - Data is not available for type HTML Format
[2021/1/26 15:40:04.613 - Clip.cpp 561] End enumerating over supported types, Count: 6
[2021/1/26 15:40:04.615 - CopyThread.cpp 111] LoadFromClipboard - After
[2021/1/26 15:40:04.619 - MainFrm.cpp 877] Start of function OnClipboardCopied, adding clip to thread for processing
[2021/1/26 15:40:04.619 - CopyThread.cpp 154] OnClipboardChange - End
[2021/1/26 15:40:04.621 - MainFrmThread.cpp 26] Adding clip to thread for save to db
[2021/1/26 15:40:04.622 - MainFrm.cpp 885] End of function OnClipboardCopied
[2021/1/26 15:40:04.622 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 2 - Name CMainFrmThread
[2021/1/26 15:40:04.627 - MainFrmThread.cpp 134] SaveCopyClips Before AddToDb
[2021/1/26 15:40:04.630 - Clip.cpp 2071] AddToDB - Start
[2021/1/26 15:40:04.631 - Clip.cpp 2082] AddToDB - while(pos), Start Remaining 1
[2021/1/26 15:40:04.634 - Clip.cpp 1473] GetNewOrder, Id: 0, parentId: -1, CurrentMax: 52127.000000, CurrentDesc: [2021/1/26 15:36:38.597, NewMax: 52128.000000
[2021/1/26 15:40:04.665 - Clip.cpp 939] Added clip to main table, Id: 24868, ParentId: -1 Desc: [2021/1/26 15:38:06.121 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 4 - Name CMainFrmThread

[2021/1/26 15:38:06.146 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 4 - Name: 8944240

[2021/1/26 15:39:06.113 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 4 - Name CMainFrmThread

[2021/1/26 15:39:06.123 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 4 - Name: 8944240

, Order: 52128.000000, GroupOrder: 0.000000
[2021/1/26 15:40:04.691 - Clip.cpp 1042] Added ClipData to DB, Id: 58711, ParentId: 24868 Type: CF_UNICODETEXT, size: 906
[2021/1/26 15:40:04.720 - Clip.cpp 1042] Added ClipData to DB, Id: 58712, ParentId: 24868 Type: CF_TEXT, size: 453
[2021/1/26 15:40:04.721 - Clip.cpp 2102] AddToDB - while(pos), End Remaining 0, save count: 1
[2021/1/26 15:40:04.723 - Clip.cpp 2105] AddToDB - Start, count: 1
[2021/1/26 15:40:04.725 - MainFrmThread.cpp 138] SaveCopyclips After AddToDb, Count: 1
[2021/1/26 15:40:04.726 - MainFrmThread.cpp 144] SaveCopyclips After AddToDb, Id: 24868 Before OnCopyCopyCompleted
[2021/1/26 15:40:04.730 - QPasteWnd.cpp 1230] OnRefreshView - Start
[2021/1/26 15:40:04.730 - MainFrmThread.cpp 148] SaveCopyclips After AddToDb, Id: 24868 After OnCopyCopyCompleted
[2021/1/26 15:40:04.732 - QPasteWnd.cpp 1260] OnRefreshView - End - Count: 0, Action: Cleared Items
[2021/1/26 15:40:04.732 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 2 - Name: 8944240
[2021/1/26 15:40:06.136 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 4 - Name CMainFrmThread
[2021/1/26 15:40:06.147 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 4 - Name: 8944240
[2021/1/26 15:41:06.143 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 4 - Name CMainFrmThread
[2021/1/26 15:41:06.153 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 4 - Name: 8944240
[2021/1/26 15:41:38.120 - MainFrm.cpp 435] Pasting clip from global shortcut, clipId: 24816
[2021/1/26 15:41:38.127 - OleClipSource.cpp 956] Start of put format on clipboard
[2021/1/26 15:41:38.131 - OleClipSource.cpp 999] Setting clipboard type: CF_TEXT to the clipboard
[2021/1/26 15:41:38.136 - OleClipSource.cpp 999] Setting clipboard type: CF_UNICODETEXT to the clipboard
[2021/1/26 15:41:38.139 - OleClipSource.cpp 1009] End of put format on clipboard
[2021/1/26 15:41:38.143 - ProcessPaste.cpp 153] start of MarkAsPasted
[2021/1/26 15:41:38.162 - ProcessPaste.cpp 171] End of MarkAsPasted
[2021/1/26 15:41:38.167 - ProcessPaste.cpp 63] Sending Paste to active window
[2021/1/26 15:41:38.162 - ProcessPaste.cpp 181] Start of MarkAsPastedThread
[2021/1/26 15:41:38.171 - ClipboardViewer.cpp 244] OnClipboardChange - End
[2021/1/26 15:41:38.174 - ProcessPaste.cpp 225] Setting clipId: 24816, order: 52129.000000
[2021/1/26 15:41:38.210 - ExternalWindowTracker.cpp 254] Sending paste to app notepad++.exe key stroke: ^{VKEY86}, SeDelay: 100
[2021/1/26 15:41:38.213 - UAC_Helper.cpp 28] I'm running elevated: 0, They are running elevated: 0, PASTE AS ADMIN: 0
[2021/1/26 15:41:38.234 - ProcessPaste.cpp 272] End of MarkAsPastedThread
[2021/1/26 15:41:38.243 - ExternalWindowTracker.cpp 285] Post sending paste

**Comment by caldertron** [comment]
I deleted my other comment because it was a long wall of log text, but the gist of it is that there is a line "Pasting clip from global shortcut, clipId: ...." but ONLY ONE, which I assume is CTRL+F7 (the one that works).

**Comment by scottbrogden-iheartmedia** [comment]
Sounds like something on your machine is stealing ctrl-f8 if Ditto isn't getting called when it's pressed.  I searched around to try and find something that we help identity what app is doing this, didn't find anything.  

Only thing I can think of is to shut down as many processes you can to see if it starts working.  Use the task manager to kill as many as you can.

**Comment by caldertron** [comment]
both CTRL+F7 & CTRL+F8 have started working in the past couple of days. Next time it stops I'll make note of what's open and see if I can figure it out!

**Comment by caldertron** [comment]
This issue has reared up again. I tried to change the text assigned to CTRL+F7 to be at CTRL+F6 and now I can no longer use either CTRL+F7 or CTRL+F6. When I select the clip from clipboard manager and click on the saved entry, it pastes all lines properly. I tried to restore the clip that had previously been assigned to CTRL+F7, but it also no longer works.

I do not believe another process is stealing CTRL+F6 & F7, since CTRL+F7 worked properly up until I tried to change it. I tried nevertheless to kill all working processes except ditto and notepad and still was not able to make either work. Interestingly, for a time CTRL+F6 pasted what was stored in CTRL+F8, but it seemed to stop on its own. 

I noticed that the one working clip has a Copy Format of 'Rich Text Format, 1204', whereas the others all have a different Copy Format (CF_TEXT, 786 and CF_UNICODETEXT, 1572). Is there a way to force the non-working clips to use that Copy Format? 

Each of the clips I am attempting to store is fairly long - about 9-11 lines. Could that be affecting it?
